src/components/organisms/AssetActions/Edit/FormEditComputeDataset.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ChangeEvent, ReactElement, useEffect } from 'react'
+import { Field, Form, FormikContextType, useFormikContext } from 'formik'
+import Button from '../../../atoms/Button'
+import Input from '../../../atoms/Input'
+import { useOcean } from '../../../../providers/Ocean'
+import { useWeb3 } from '../../../../providers/Web3'
+import { FormFieldProps } from '../../../../@types/Form'
+import { ServiceComputePrivacy } from '@oceanprotocol/lib'
+import { AssetSelectionAsset } from '../../../molecules/FormFields/AssetSelection'
+import slugify from 'slugify'
+import stylesIndex from './index.module.css'
+import styles from './FormEditMetadata.module.css'
+
+export default function FormEditComputeDataset({
+  data,
+  title,
+  setShowEdit,
+  values,
+  algorithmList
+}: {
+  data: FormFieldProps[]
+  title: string
+  setShowEdit: (show: boolean) => void
+  values: ServiceComputePrivacy
+  algorithmList: AssetSelectionAsset[]
+}): ReactElement {
+  const { accountId } = useWeb3()
+  const { ocean } = useOcean()
+  const {
+    isValid,
+    validateField,
+    setFieldValue
+  }: FormikContextType<ServiceComputePrivacy> = useFormikContext()
```

Comment:
or even simpler, you can get `values` & `setValues` & `initialValues` exported from `useFormikContext()`. So we do not need to pass them all as props and just work in this component. Then in this component only work with `setValues()` instead of `values.push`